nntrainer/src/fc_layer.cpp

```diff
@@ -149,49 +153,61 @@ Tensor FullyConnectedLayer::forwarding(Tensor in, Tensor output, int &status) {
     y = y.apply(activation);
   }
 
+  Tensor l;
+
   switch (cost) {
   case COST_MSR: {
-    Tensor sub = y2.subtract(y);
-    Tensor l = (sub.multiply(sub)).sum().multiply(0.5);
+    // y2 <- y2 - y;
+    y2.subtract_i(y);
+
```

Comment:
is y2 overwritten? I think it should be as it is. it is going to be used line #172.

Reply:
Yes. It is overwritten.
I checked that if code hits `:161`, it would not hit `:172`. 
So, I think it is safely done for this line.

However,  IMHO, this kind of use(eg. `a.subtract_i(b)`) shouldn't be allowed for safer coding.
For performance point of view, this kind of operation should be allowed.

Could you please give a opinion about this kind of (rather unsafe) optimization?

Reply:
Oh. yes. I will. and about this line, you are right. Thanks.